File: Algorithm/CountItemsMatchingaRule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        return count_if(begin(items), end(items), [&](const auto& item) {
           return (ruleKey == "type" && item[0] == ruleValue
             || ruleKey == "color" && item[1] == ruleValue
             || ruleKey == "name" && item[2] == ruleValue) ;
        });
    }
};

File: Algorithm/CountItemsMatchingaRule.java
// Solution: Brute Force
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int ans = 0;
        for (List<String> item : items) {
            ans += ((ruleKey.equals("type") && item.get(0).equals(ruleValue)) ||
                    (ruleKey.equals("color") && item.get(1).equals(ruleValue)) ||
                    (ruleKey.equals("name") && item.get(2).equals(ruleValue))) ? 1 : 0;
        }
        return ans;
    }
}
